test_unstructured/partition/test_common.py

```diff
@@ -405,3 +417,12 @@ def test_set_element_hierarchy_custom_rule_set():
     assert (
         elements[5].metadata.parent_id == elements[4].id
     ), "FigureCaption should be child of Title 2"
+
+
+def test_document_to_element_list_sets_category_depth_titles():
+    layout_with_hierarchies = MockDocumentLayout()
+    elements = document_to_element_list(layout_with_hierarchies)
+    assert elements[0].metadata.category_depth == 1
+    assert elements[1].metadata.category_depth == 2
+    assert elements[2].metadata.category_depth is None
```

Comment:
What's the reasoning for this category_depth being None vs being set to 0?

Reply:
`elements[2]` is a `Text` type. The default value for the `category_depth` metadata for all elements is None. The method `to_dict` of `ElementMetadata` returns all metadata that is not None ([here](https://github.com/Unstructured-IO/unstructured/blob/ab29de8dbdd3e75cbf0d61b5515efec322547a9b/unstructured/documents/elements.py#L194)). Many tests count on this behaviour and expect empty metadata so the Element method `to_dict` returns empty metadata and the json contains only the Elements types + text instead of a list of elements all with 0 value in `category_depth`. I did consider this before so I get you but after discussion with @qued, we thought the None value was more convenient.